spec/requests/users_controller_spec.rb

```diff
@@ -1,51 +1,2402 @@
 require 'rails_helper'
 
-RSpec.describe UsersController do
+describe UsersController do
   let(:user) { Fabricate(:user) }
 
-  def honeypot_magic(params)
-    get '/u/hp.json'
-    json = JSON.parse(response.body)
-    params[:password_confirmation] = json["value"]
-    params[:challenge] = json["challenge"].reverse
-    params
+  describe '#activate_account' do
+    let(:token) { "asdfasdf" }
+    before do
+      UsersController.any_instance.stubs(:honeypot_or_challenge_fails?).returns(false)
+    end
+
+    context 'invalid token' do
+      it 'return success' do
+        put "/u/activate-account/#{token}"
+        expect(response).to be_success
+        expect(flash[:error]).to be_present
+      end
+    end
+
+    context 'valid token' do
+      let(:user) { Fabricate(:user) }
+
+      context 'welcome message' do
+        before do
+          EmailToken.expects(:confirm).with("#{token}").returns(user)
+        end
+
+        it 'enqueues a welcome message if the user object indicates so' do
+          user.send_welcome_message = true
+          user.expects(:enqueue_welcome_message).with('welcome_user')
```

Comment:
We should remove this stub too and tests for the actual job that is being enqueued. You'll have to set `SiteSetting.queue_jobs = true` and you'll be able to access the jobs via `Jobs::EnqueueWelcomeMessage.jobs`